diagnostic_feedback/tests/integration/base_test.py

```diff
@@ -0,0 +1,166 @@
+
+loader = ResourceLoader(__name__)
+
+
+class DiagnosticFeedbackBaseTest(SeleniumXBlockTest):
+    """
+    Base class for integration tests.
+    Scenarios can be loaded and edited on the fly.
+    """
+
+    def _get_previous_button(self):
+        return self.browser.find_elements_by_xpath("//ul[@role='menu']/li")[0]
+
+    def _get_previous_button_link(self):
+        return self.browser.find_element_by_xpath("//ul[@role='menu']/li/a[@href='#previous']")
+
+    def _get_next_button(self):
+        return self.browser.find_elements_by_xpath("//ul[@role='menu']/li")[1]
+
+    def _get_next_button_link(self):
+        return self.browser.find_element_by_xpath("//ul[@role='menu']/li/a[@href='#next']")
+
+    def wait_until_has_class_disabled(self, elem):
+        """ Need to wait until the given element has disabled class """
+
+        wait = WebDriverWait(elem, self.timeout)
+        wait.until(lambda e: 'disabled' in elem.get_attribute('class'),
+                   u"{} should be disabled".format(elem.text))
+
+    def _get_final_result_text(self):
+        return self.browser.find_element_by_css_selector('div.response_body').text
+
+    def _get_final_results(self):
+        return self.browser.find_elements_by_css_selector('div.response_body div.result')
+
+    def load_scenario(self, xml_file, params=None, load_immediately=True, view='student_view'):
+        """
+        Given the name of an XML file in the xml_templates folder, load it into the workbench.
+        """
+        params = params or {}
+        scenario = loader.render_template("xml_templates/{}".format(xml_file), params)
+        self.set_scenario_xml(scenario)
+        if load_immediately:
+            return self.go_to_view(view)
+
+    def load_student_view(self, xml_file, params=None):
+        """
+        Load student view in workbench
+        """
+        self.load_scenario("student_{}.xml".format(xml_file), params=params)
+
+    def load_studio_view(self, xml_file, params=None):
+        """
+        Load studio view in workbench
+        """
+        self.load_scenario("studio_{}.xml".format(xml_file), params=params, view='studio_view')
+
+
+class StudentBaseTest(DiagnosticFeedbackBaseTest):
+
+    def _get_startover_button(self):
+        return self.browser.find_elements_by_xpath("//ul[@role='menu']/li")[3]
+
+    def _get_startover_button_link(self):
+        return self.browser.find_element_by_xpath("//ul[@role='menu']/li/a[@href='#cancel']")
+
+    def _get_visible_choices(self):
+        """
+        Return one OR all visible choices
+        """
+        return [c for c in self.browser.find_elements_by_xpath("//input[@type='radio']") if c.is_displayed()]
+
+    def _is_any_choice_selected(self):
+        return any(choice.is_selected() for choice in self._get_visible_choices())
+
+    def _verify_question(self, question_order=0, next_btn=None, back_btn=None, choice_idx=0):
+        self.assertFalse(self._is_any_choice_selected(), False)
```

Comment:
@jareerahsan I should have been more specific: One of the advantages of (and reasons for) using `assertFalse` and `assertTrue` is that you don't have to provide a value to compare against. They are used like this:

```python
self.assertFalse(self._is_any_choice_selected())
self.assertTrue(self._is_any_choice_selected())
```

The reason why this line is not throwing any errors is because `assertFalse` and `assertTrue` do take a second argument which is optional and lets you specify an error message that will be printed when the assertion fails. (For reference, the [`unittest.TestCase`](https://docs.python.org/2/library/unittest.html#unittest.TestCase) docs include a handy list of available assertion methods and how to use them.)